src/plugins/kibana/server/routes/api/index_pattern_endpoints.js

```diff
@@ -0,0 +1,258 @@
+      return Boom.serverTimeout(error);
+    } else if (error instanceof esErrors.Conflict || _.contains(error.message, 'index_template_already_exists')) {
+      return Boom.conflict(error);
+    } else if (error instanceof esErrors[403]) {
+      return Boom.forbidden(error);
+    } else if (error instanceof esErrors.NotFound) {
+      return Boom.notFound(error);
+    } else if (error instanceof esErrors.BadRequest || error instanceof TypeError) {
+      return Boom.badRequest(error);
+    } else {
+      return error;
+    }
+  };
+
+  server.route({
+    path: '/api/kibana/index_patterns',
+    method: 'GET',
+    handler: function (req, reply) {
+      const callWithRequest = server.plugins.elasticsearch.callWithRequest;
+
+      const params = {
+        index: '.kibana',
+        type: 'index-pattern',
+        body: {
+          query: {
+            match_all: {}
+          }
+        }
+      };
+
+      callWithRequest(req, 'search', params)
+        .then(function parseResults(results) {
+          const hits = results.hits.hits;
+          return _.map(hits, (patternHit) => {
+            patternHit._source.fields = JSON.parse(patternHit._source.fields);
+            return patternHit._source;
+          });
+        })
+        .then((patterns) => {
+          return Promise.map(patterns, (pattern) => {
+            return getMappings(pattern.title, req).catch(() => {
+              return {};
+            }).then((mappings) => {
+              return stitchPatternAndMappings(pattern, mappings);
+            });
+          });
+        })
+        .then(removeDeprecatedFieldProps)
+        .then((patterns) => {
+          reply(patterns);
+        }, function (error) {
+          reply(handleESError(error));
+        });
+    }
+  });
+
+  server.route({
+    path: '/api/kibana/index_patterns/{id}',
+    method: 'GET',
+    handler: function (req, reply) {
+      const callWithRequest = server.plugins.elasticsearch.callWithRequest;
+      let pattern = req.params.id;
+
+      const params = {
+        index: '.kibana',
+        type: 'index-pattern',
+        id: req.params.id
+      };
+
+      Promise.join(
+        callWithRequest(req, 'get', params)
+          .then((result) => {
+            result._source.fields = JSON.parse(result._source.fields);
+            return result._source;
+          }),
+        getMappings(pattern, req),
+        stitchPatternAndMappings)
+        .then(removeDeprecatedFieldProps)
+        .then(function (pattern) {
+          reply(_.isArray(pattern) ? pattern[0] : pattern);
+        }, function (error) {
+          reply(handleESError(error));
+        });
+    }
+  });
+
+  server.route({
+    path: '/api/kibana/index_patterns',
+    method: 'POST',
+    handler: function (req, reply) {
+      if (_.isEmpty(req.payload)) { return reply(Boom.badRequest('Payload required')); }
+      const validation = Joi.validate(req.payload, indexPatternSchema.post);
```

Comment:
The hapi [route options](http://hapijs.com/api/8.8.1#route-options) accept validation configuration that is perfect for this type of work.